src/LaunchDarkly.ClientSdk/LdClient.cs

```diff
@@ -226,14 +234,43 @@ public sealed class LdClient : ILdClient
 
         void Start(TimeSpan maxWaitTime)
         {
+            if (maxWaitTime >= ExcessiveInitWaitTime)
+            {
+                _log.Warn(ExcessiveInitWaitTimeWarning, maxWaitTime, ExcessiveInitWaitTime);
+            }
+
             var success = AsyncUtils.WaitSafely(() => _connectionManager.Start(), maxWaitTime);
             if (!success)
             {
-                _log.Warn("Client did not successfully initialize within {0} milliseconds.",
-                    maxWaitTime.TotalMilliseconds);
+                _log.Warn(DidNotInitializeTimelyWarning, maxWaitTime.TotalMilliseconds);
             }
         }
 
+        /// <summary>
+        /// Starts the client and waits up to the wait time to initialize feature flags.  If offline,
+        /// returns immediately.
+        /// </summary>
+        /// <param name="maxWaitTime">the maximum length of time to wait for the client to initialize</param>
+        async Task StartAsync(TimeSpan  maxWaitTime)
+        {
+            if (maxWaitTime >= ExcessiveInitWaitTime)
+            {
+                _log.Warn(
+                    ExcessiveInitWaitTimeWarning,
+                    maxWaitTime, ExcessiveInitWaitTime);
+            }
+
+            var startTask = _connectionManager.Start();
+            var completedTask = await Task.WhenAny(startTask, Task.Delay(maxWaitTime));
```

Comment:
How does this `!=` logic work? (just haven't seen it before with tasks)

Reply:
The whenAny returns the task that completed and if it isn't the start task then it must be the delay task.